app/src/androidTest/java/com/ncnf/views/fragments/group/GroupFragmentTest.java

```diff
@@ -0,0 +1,204 @@
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static androidx.test.platform.app.InstrumentationRegistry.getInstrumentation;
+import static com.ncnf.utilities.StringCodes.EVENTS_COLLECTION_KEY;
+import static com.ncnf.utilities.StringCodes.FULL_NAME_KEY;
+import static com.ncnf.utilities.StringCodes.USERS_COLLECTION_KEY;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.when;
+
+@UninstallModules(CurrentUserModule.class)
+@HiltAndroidTest
+public class GroupFragmentTest {
+
+    static private final User user1 = Mockito.mock(User.class);
+    static private final GroupRepository repository1 = Mockito.mock(GroupRepository.class);
+    static private final UserRepository repository2 = Mockito.mock(UserRepository.class);
+
+    static private UUID gUuid = UUID.randomUUID();
+    static private final Group g = new Group("u1",gUuid, "Group Test", LocalDateTime.now(), new GeoPoint(-0.03, -0.03), "random address", SocialObject.Type.OTHER, new ArrayList<>(), "description here", new ArrayList<>());
+
+    private HiltAndroidRule hiltRule = new HiltAndroidRule(this);
+
+    @InjectMocks
+    private final ActivityScenarioRule scenario = new ActivityScenarioRule<>(GroupActivity.class);
+
+    @Rule
+    public RuleChain testRule = RuleChain.outerRule(hiltRule).around(scenario);
+
+    @BindValue
+    public User user = user1;
+
+    @BindValue
+    public GroupRepository repository = repository1;
+
+    @BindValue
+    public UserRepository userRepository = repository2;
+
+    @BeforeClass
+    static public void injects() {
+        when(user1.loadUserFromDB()).thenReturn(CompletableFuture.completedFuture(user1));
+        ArrayList<String> l = new ArrayList<>();
+        l.add(gUuid.toString());
+
+        ArrayList<Group> l2 = new ArrayList<>();
+        l2.add(g);
+
+        when(repository2.loadUser("u1")).thenReturn(CompletableFuture.completedFuture(user1));
+
+        when(user1.getParticipatingGroupsIds()).thenReturn(new ArrayList<>());
+        when(user1.getOwnedGroupsIds()).thenReturn(l);
+
+        when(repository1.loadMultipleGroups(l)).thenReturn(CompletableFuture.completedFuture(l2));
+
+        when(repository1.loadGroup(gUuid.toString())).thenReturn(CompletableFuture.completedFuture(g));
+
+        when(repository2.getUserFullName("u1")).thenReturn(CompletableFuture.completedFuture("John"));
+
+    }
+
+    @Before
+    public void setup() {
+
+        Intents.init();
+    }
+
+
+    @After
+    public void cleanup(){
+        Intents.release();
+    }
```

Comment:
Useless if you don't use Intents.indented() which is in one test. Maybe better move this in the test.